### ISSUE TITLE
Feature Request: Add permanent_deletion_time_in_days Argument to aws_acmpca_certificate_authority

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#4994]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The ACM Private Certificate Authority API recently added a new parameter when calling `DeleteCertificateAuthority`: `PermanentDeletionTimeInDays`

Its description:

```
    // The number of days to make a CA restorable after it has been deleted. This
    // can be anywhere from 7 to 30 days, with 30 being the default.
    PermanentDeletionTimeInDays *int64 `min:"7" type:"integer"`
```

This appears to be similar to the [`aws_kms_key` resource `deletion_window_in_days` argument](https://www.terraform.io/docs/providers/aws/r/kms_key.html#deletion_window_in_days)

### New or Affected Resource(s)

* `aws_acmpca_certificate_authority`

### Potential Terraform Configuration

```hcl
# Example implementation, may change after development
resource "aws_acmpca_certificate_authority" "example" {
  # ... other configuration ...
  permanent_deletion_time_in_days = 7
}
```

### References

* https://docs.aws.amazon.com/sdk-for-go/api/service/acmpca/#DeleteCertificateAuthorityInput

**Comment by vnguyen7** [comment]
Are there any resources dependent on this aws_acmpca_certificate_authority ? I've been working on it a few days but the example still got rejected

**Comment by bflad** [comment]
The new `permanent_deletion_time_in_days` argument for the `aws_acmpca_certificate_authority` resource has been merged and will release with version 2.5.0 of the Terraform AWS Provider later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 2.5.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#250-april-05-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
